python_modules/dagster/dagster/_core/execution/submit_asset_runs.py

```diff
@@ -142,47 +148,58 @@ def _create_asset_run(
                 workspace=workspace,
                 run_request_execution_data_cache=run_request_execution_data_cache,
             )
-            check_for_debug_crash(debug_crash_flags, "EXECUTION_PLAN_CREATED")
-            check_for_debug_crash(debug_crash_flags, f"EXECUTION_PLAN_CREATED_{run_request_index}")
 
-            # retry until the execution plan targets the asset selection
-            if _execution_plan_targets_asset_selection(
-                execution_data.external_execution_plan.execution_plan_snapshot,
+        except (DagsterInvalidSubsetError, DagsterUserCodeProcessError):
+            logger.warning(
+                "Error while generating the execution plan, possibly because the code server is out of sync with the daemon. Pausing run submission for "
+                f"{RELOAD_WORKSPACE_INTERVAL} to allow the execution plan to rebuild with the updated workspace.",
+                exc_info=sys.exc_info(),
+            )
+            should_retry = True
+
+        check_for_debug_crash(debug_crash_flags, "EXECUTION_PLAN_CREATED")
+        check_for_debug_crash(debug_crash_flags, f"EXECUTION_PLAN_CREATED_{run_request_index}")
+
+        if (not should_retry) and (
+            not _execution_plan_targets_asset_selection(
+                check.not_none(execution_data).external_execution_plan.execution_plan_snapshot,
                 check.not_none(run_request.asset_selection),
-            ):
-                external_job = execution_data.external_job
-                external_execution_plan = execution_data.external_execution_plan
-                partitions_def = execution_data.partitions_def
-
-                run = instance.create_run(
-                    job_snapshot=external_job.job_snapshot,
-                    execution_plan_snapshot=external_execution_plan.execution_plan_snapshot,
-                    parent_job_snapshot=external_job.parent_job_snapshot,
-                    job_name=external_job.name,
-                    run_id=run_id,
-                    resolved_op_selection=None,
-                    op_selection=None,
-                    run_config={},
-                    step_keys_to_execute=None,
-                    tags=run_request.tags,
-                    root_run_id=None,
-                    parent_run_id=None,
-                    status=DagsterRunStatus.NOT_STARTED,
-                    external_job_origin=external_job.get_external_origin(),
-                    job_code_origin=external_job.get_python_origin(),
-                    asset_selection=frozenset(run_request.asset_selection),
-                    asset_check_selection=None,
-                    asset_job_partitions_def=partitions_def,
-                )
+            ),
+        ):
+            logger.warning(
```

Comment:
Just so I am understanding this correctly, what was happened here is that there is an explicit asset selection on the run request, and then the generated execution plan is out of sync with that (i.e. has a different asset selection?)

In that case what does waiting do for us? Why will the retry suceed?

Reply:
Waiting and retrying will succeed because the daemon periodically refreshes its copy of the snapshots that it used to determine which job needed to be launched for these assets, so they will become in sync again

Reply:
Typically it's not that it has a different asset selection, but rather than it is looking for those assets on the entirely incorrect job due to the way we index base asset jobs

Reply:
Ok.

What is the daemon refresh schedule?

It seems odd and wobbly that we are waiting for some magical, unrelated process to do something rather than just synchronously doing it when we know we need to do it.